scripts/feature_tests/upgrade/upgrade_common.sh

```diff
@@ -230,21 +232,23 @@ function worker_has_correct_replicas() {
     replicas="${1}"
     wr_replicas=0
     if [[ "${replicas}" -eq 0 ]]; then
-        echo "Waiting for all replicas of worker nodes to leave the cluster"
+      echo "Waiting for all replicas of worker nodes to leave the cluster"
     else
-        echo "Waiting for all replicas of worker nodes to join the cluster"
+      echo "Waiting for all replicas of worker nodes to join the cluster"
     fi
 
     for i in {1..1800}; do
-        wr_replicas=$(kubectl get nodes | awk 'NR>1' | grep -vc master)
+        wr_replicas=$(kubectl get bmh -n metal3 | grep -i provisioned | grep -c worker)
```

Comment:
namespace parameterization

Reply:
shall be fixed. Not before merge because the CI is so difficult

Reply:
fixed in https://github.com/metal3-io/metal3-dev-env/pull/497